File: 56. Merge Intervals/main.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:

        intervals.sort(key=lambda x:x[0])

        answers=[]


        for interval in intervals:
            if not answers or answers[-1][1] < interval[0]:
                answers.append(interval)

            else:
                answers[-1][1]=max(answers[-1][1],interval[1])


        return answers;
        

File: 56. Merge Intervals/main.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {

        std::sort(intervals.begin(), intervals.end());

        std::vector<std::vector<int>> answers;

        int n = intervals.size();
        for (int i = 0; i < n; i++) {
            auto start = intervals[i][0];

            int maxEnd=intervals[i][1];


            while (i < n - 1 && maxEnd >= intervals[i + 1][0]) {
                maxEnd=std::max(intervals[i+1][1],maxEnd);
                i++;
            }

            auto end = maxEnd;
            answers.push_back({start, end});
        }

        return answers;
    }
};